capstone/src/main/java/com/google/sps/servlets/GroupPostDataServlet.java

```diff
@@ -69,66 +68,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(new Gson().toJson(postsRes));
   }
 
-  public Post postEntity(Entity entity) {
-    Post userPost = Post.getPostEntity(entity);
-    // Create list of comment objects from list of comment entities 
-    if (entity.getProperty("comments") != null) {
-      createCommentObjectList(userPost.getComments(), entity);
-    }
-    return userPost;
-  }
-
-  private void createCommentObjectList(ArrayList<Comment> comments, Entity entity) {
-    ArrayList<EmbeddedEntity> commentEntitys = (ArrayList<EmbeddedEntity>) entity.getProperty("comments");
-    for (EmbeddedEntity comment: commentEntitys) {
-      comments.add(Comment.getCommentEntity(comment));
-    }
-  }
-
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Receives submitted post 
-    long timestamp = System.currentTimeMillis();
     String authorName = "Jane Doe";
     String postText = request.getParameter("post-input");
     String challengeName = "Challenge Name";
     String img = getUploadedFileUrl(request, "image");
-    ArrayList<String> likes = new ArrayList<>();
+    HashSet<String> likes = new HashSet<>();
     ArrayList<Comment> comments = new ArrayList<>();
 
     // Creates entity with submitted data and add to database
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(createPostEntity(timestamp, authorName, postText, challengeName, img, likes, comments));
+    Post post = new Post(0, authorName, postText, comments, challengeName, System.currentTimeMillis(), img, likes);
```

Comment:
This looks like every post has an ID of 0, is that intended?

Reply:
I'm using the 0 as a placeholder id. Right after I'm creating a post entity, in which the id will be replaced with an id generated by the database.